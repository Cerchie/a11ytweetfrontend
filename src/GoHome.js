import React from 'react'

function GoHome() {
    return (
        <>
            <p style={{ margin: '1rem' }}>Success!</p>
            <a href="/" style={{ margin: '1rem' }}>
                Return home.{' '}
            </a>
        </>
    )
}

export default GoHome
