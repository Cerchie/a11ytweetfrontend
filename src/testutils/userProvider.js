import React from 'react'
import UserContext from '../UserContext'

const demoUser = {
    username: 'testuser',
}

const UserProvider = ({ children, currentUser = demoUser }) => (
    <UserContext.Provider value={{ currentUser }}>
        {children}
    </UserContext.Provider>
)

export { UserProvider }
