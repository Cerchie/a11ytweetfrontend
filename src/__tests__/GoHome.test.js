import { render, screen } from '@testing-library/react'
import React from 'react'
import GoHome from '../GoHome'
import { MemoryRouter } from 'react-router'
import { UserProvider } from '../testutils/userProvider'

test('renders app', () => {
    render(<GoHome />)
})

it('matches snapshot', function () {
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider>
                <GoHome />
            </UserProvider>
        </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot()
})
