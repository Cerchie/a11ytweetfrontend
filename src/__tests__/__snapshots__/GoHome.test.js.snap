// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <p
    style="margin: 1rem;"
  >
    Success!
  </p>
  <a
    href="/"
    style="margin: 1rem;"
  >
    Return home. 
  </a>
</DocumentFragment>
`;
